src/Compilers/CSharp/Portable/Lowering/LocalRewriter/DelegateCacheRewriter.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp;
+
+internal sealed partial class DelegateCacheRewriter
```

Comment:
>DelegateCacheRewriter [](http://example.com/codeflow?start=30&length=21)

Consider adding a doc comment briefly describing criteria for caching and the strategy (type level/method level, etc.)

Reply:
Added. Please take a look.

Reply:
This type is renamed to `DelegateCreationRewriter`.